ibis/backends/bigquery/registry.py

```diff
@@ -490,6 +491,60 @@ def _array_repeat(t, op):
     return f"ARRAY(SELECT {arg}[SAFE_ORDINAL({idx})] FROM UNNEST({series}) AS i)"
 
 
+def _neg_idx_to_pos(array, idx):
+    return f"IF({idx} < 0, ARRAY_LENGTH({array}) + {idx}, {idx})"
+
+
+def _array_slice(t, op):
+    arg = t.translate(op.arg)
+    cond = [f"index >= {_neg_idx_to_pos(arg, t.translate(op.start))}"]
+    if op.stop:
+        cond.append(f"index < {_neg_idx_to_pos(arg, t.translate(op.stop))}")
+    return (
+        f"ARRAY("
+        f"SELECT el "
+        f"FROM UNNEST({arg}) AS el WITH OFFSET index "
+        f"WHERE {' AND '.join(cond)}"
+        f")"
+    )
+
+
+def _capitalize(t, op):
+    return f"CONCAT(UPPER(SUBSTR({t.translate(op.arg)}, 1, 1)), SUBSTR({t.translate(op.arg)}, 2))"
+
+
+def _translate(t, op):
```

Comment:
I'm not sure we have sufficient test coverage for this function, so let's leave it out. This API is supposed to be modeled after Python's [`str.translate`](https://docs.python.org/3/library/stdtypes.html#str.translate) method, but to be honest I don't think we have a non-trivial test for this API.

What do you think about implementing this operation in a follow-up PR and leaving it out on this one for now?

Reply:
Done